apps/files/src/components/Files-App.vue

```diff
@@ -1,257 +1,339 @@
-					this.loading = false;
-					this.resetSelect();
-				}, delay)
-
-			}).fail(function (e) {
-				console.log(e);
-				UIkit.notification({
-					message: e.statusText,
-					status: 'danger',
-					pos: 'top-center'
-				});
-			})
-		},
-		singleSelect (item) {
-			this.selected = [item];
-		},
-		resetSelect () {
-			this.selected = [];
-		},
-		multiSelect (item) {
-			if (this.isChecked(item))
-				_.remove(this.selected, item);
-			else
-				this.selected.push(item);
-		},
-		isChecked(item) {
-			return _.includes(this.selected, item);
-		}
-	},
-	watch: {
-		item () {
-			this.loadFolder()
-		}
-	},
-	computed: {
-		item () {
-			return this.$route.params.item.toLowerCase();
-		},
-		typeOfFolder () {
-			return _.filter(this.files, ['extension' , false ])
-		},
-		typeOfFile (showHidden) {
-			showHidden = (typeof showHidden !== 'undefined') ?  showHidden : false;
-			return _.filter(this.files, 'extension')
-		}
-	}
-}
+    import Mixins from '../mixins';
+    import FileDetails from './File-Details.vue';
+
+    const remove = require('lodash/remove');
+    const includes = require('lodash/includes');
+    const filter = require('lodash/filter');
+
+    export default {
+        mixins: [Mixins],
+        components: {FileDetails},
+        data() {
+            return {
+                loading: false,
+                filterBy: {
+                    files: true,
+                    folder: true,
+                    hidden: false
+                },
+                path: [],
+                files: [],
+                selected: [],
+                self: {}
+            }
+        },
+        mounted() {
+            OC.$bus.on('phoenix:user-logged-in', () => {
+                this.loadFolder();
+            })
+        },
+        methods: {
+            goto(e) {
+                this.$route.push()
+            },
+            toggleFavorite(item) {
+                this.files[item].stared = (!this.files[item].stared);
+            },
+            routerLink(itemPath) {
+                this.$router.push({
+                    name: 'files',
+                    params: {
+                        item: itemPath
+                    }
+                })
+            },
+            loadFolder() {
+                this.loading = true;
+
+                let absolutePath = this.$route.params.item;
+                if(this.$route.params.item === 'home'){
+                    absolutePath = '/';
+                    this.path = []
+                }else{
+                    let pathSplit  = absolutePath.split('/');
+                    pathSplit = pathSplit.slice(1, pathSplit.length - 1);
+                    this.path = pathSplit
+                }
+
+                // List all files
+                OC.$client.files.list(absolutePath).then(files => {
+                    // Remove the root element
+                    files = files.splice(1);
+
+                    let delay = Math.random() * (1000 - 200) + 200;
```

Comment:
I wrote this to demonstrate a delay + the loading animation. We can gladly get rid of that :-)